### DISCUSSION TITLE
how to make sure capacity of vehicle would not control number of nodes delivered ?

### DISCUSSION THREAD

**Discussion by orangiee** [google/or-tools#3043] [Routing (and legacy CP) questions]
Hi， I'm working on a code of vrp_reload.py. And I found that the number of nodes that a vehicle can do is controlled by capacity of vehicle, for example, when I set 1000 for capacity of vehicle, and at reload node, it would be -1000. so before vehicle come back to reload node, it will delivery many nodes until load value reach almost 1000, even though the distance between the nodes are very huge, it must reach 1000. How can I do to make sure that the number of nodes delivered are not controlled by capacity of vehicle. plus, I cannot set data['vehicle_max_distance'] which is not our goal, even though we want do as most nodes as possible with minimal distances. 
My question is:
1, how to make sure the distance between nodes are minimal for each vehicle, and not just to reach the capacity of vehicle to do the delivery which may forget the distances between nodes.  
2, in the code, we set 5 vehicles, and the load of route of vehicles are almost the same value, at least, they are very approximate values, for example, vehicle One have 986 for the load of route, vehicle Two have 979 for the load of route, vehicle Three have 979, vehicle Four have 999, etc. each vehicle juste try to reach the capacity of vehicle, and not consider the distance between nodes, even though I have set time windows for nodes, and for reload node and for depot node. How can make sure that the load of route  of each vehicle DON'T need to be very approximate ? 
3, can you explain `distance_dimension.SetCumulVarSoftUpperBound` and expecially how does it influce the result ? 
Thank you so much in advance !!

**Comment by Mizux** [comment]
actually while you have `-1000` for reload node as demand transit callback you have the slack var in range `[0, 1000]` which avoid to have negative value.

e.g. vehicle arrives at reload node with `450` capacity you'll have `450 + -1000 + slack == 0` i.e. solver will set slack to `550`...

**Reply by orangiee** [reply]
thanks for reply, but I don't understand. in capacity dimension, what does 'slack' represent ? what's this 550 ? and how to explain that: vehicle arrives at reload node with almost 1000 capacity each time and for each vehicle ? Thanks so much!
